recipes/mdnsresponder/all/conanfile.py

```diff
@@ -110,63 +130,74 @@ def _msvc_platform(self):
+        sln = os.path.join(self.source_folder, "mDNSResponder.sln")
         if "MD" in self.settings.compiler.runtime:
             # could use glob and replace_in_file(strict=False, ...)
-            dll_vcxproj = os.path.join(self._source_subfolder, "mDNSWindows", "DLL", "dnssd.vcxproj")
-            dllstub_vcxproj = os.path.join(self._source_subfolder, "mDNSWindows", "DLLStub", "DLLStub.vcxproj")
-            dns_sd_vcxproj = os.path.join(self._source_subfolder, "Clients", "DNS-SD.VisualStudio", "dns-sd.vcxproj")
+            dll_vcxproj = os.path.join(self.source_folder, "mDNSWindows", "DLL", "dnssd.vcxproj")
+            dllstub_vcxproj = os.path.join(self.source_folder, "mDNSWindows", "DLLStub", "DLLStub.vcxproj")
+            dns_sd_vcxproj = os.path.join(self.source_folder, "Clients", "DNS-SD.VisualStudio", "dns-sd.vcxproj")
             for vcxproj in [dll_vcxproj, dllstub_vcxproj, dns_sd_vcxproj]:
-                tools.replace_in_file(vcxproj, "<RuntimeLibrary>MultiThreaded</RuntimeLibrary>", "<RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>")
-                tools.replace_in_file(vcxproj, "<RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>", "<RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>")
+                replace_in_file(self, vcxproj, "<RuntimeLibrary>MultiThreaded</RuntimeLibrary>", "<RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>")
+                replace_in_file(self, vcxproj, "<RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>", "<RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>")
 
         # could use glob and replace_in_file(strict=False, ...)
-        dll_rc = os.path.join(self._source_subfolder, "mDNSWindows", "DLL", "dll.rc")
-        dns_sd_rc = os.path.join(self._source_subfolder, "Clients", "DNS-SD.VisualStudio", "dns-sd.rc")
+        dll_rc = os.path.join(self.source_folder, "mDNSWindows", "DLL", "dll.rc")
+        dns_sd_rc = os.path.join(self.source_folder, "Clients", "DNS-SD.VisualStudio", "dns-sd.rc")
         for rc in [dll_rc, dns_sd_rc]:
-            tools.replace_in_file(rc, "afxres.h", "winres.h")
+            replace_in_file(self, rc, "afxres.h", "winres.h")
 
         msbuild = MSBuild(self)
-        msbuild.build(sln, targets=self._msvc_targets, platforms=self._msvc_platforms, definitions=self._msvc_definitions)
+        # use Win32 instead of the default value when building x86
+        msbuild.platform = "Win32" if self.settings.arch == "x86" else msbuild.platform
+        #TODO: How to add _msvc_definitions to msbuild.build
+        msbuild.build(sln, targets=self._msvc_targets)
+
+    def generate(self):
+        if self.settings.os == "Linux":
+            tc = AutotoolsToolchain(self)
+            # fix for error: 'for' loop initial declarations are only allowed in C99 or C11 mode
+            tc.extra_cflags.append("-std=gnu99")
+            tc.generate()
+            tc = AutotoolsDeps(self)
+            tc.generate()
```

Comment:
```suggestion
            tc.generate()
        elif is_msvc(self):
            tc = MSBuildToolchain(self)
            tc.configuration = "Debug" if self.settings.build_type == "Debug" else "Release"
            tc.generate()
            tc = VCVars(self)
            tc.generate()
```

It should fix msbuild missing error. Please, don't forget to update imports too.

Reply:
Thank you very much, it works in my local  Windows.